packages/core/src/util/transactionSummaryInspector.ts

```diff
@@ -147,8 +151,34 @@ const toAssetInfoWithAmount = async (
  */
 export const transactionSummaryInspector: TransactionSummaryInspector =
   (args: TransactionSummaryInspectorArgs) => async (tx) => {
-    const { inputResolver, addresses, rewardAccounts, protocolParameters, assetProvider, dRepKeyHash } = args;
-    const resolvedInputs = await resolveInputs(tx.body.inputs, inputResolver);
+    const {
+      inputResolver,
+      addresses,
+      rewardAccounts,
+      protocolParameters,
+      assetProvider,
+      dRepKeyHash,
+      timeout,
+      logger
+    } = args;
+
+    let resolvedInputs: ResolutionResult;
+
+    try {
+      // eslint-disable-next-line sonarjs/prefer-immediate-return
+      resolvedInputs = await resolveWithTimeout(resolveInputs(tx.body.inputs, inputResolver), timeout);
+    } catch (error) {
+      if (error === 'Timeout: failed to resolve inputs in time') {
+        // Handle timeout error specifically
+        logger?.log('Error: Inputs resolution timed out');
```

Comment:
Is `logger` optional? It shouldn't be
```suggestion
        logger.log('Error: Inputs resolution timed out');
```

Reply:
✅